packages/cookies/src/response-cookies.ts

```diff
@@ -0,0 +1,110 @@
+import { Cookies } from './cookies'
+import type { CookieSerializeOptions } from 'cookie'
+import * as cookie from 'cookie'
+
+type HasHeaders = { headers: Headers }
+
+const deserializeCookie = (input: HasHeaders): string[] => {
+  const value = input.headers.get('set-cookie')
+  return value !== undefined && value !== null ? value.split(', ') : []
+}
+
+const serializeCookie = (input: string[]) => input.join(', ')
+
+type GetWithOptionsOutput = {
+  value: string | undefined
+  options: { [key: string]: string }
+}
+
+const serializeExpiredCookie = (
+  key: string,
+  options: CookieSerializeOptions = {}
+) =>
+  cookie.serialize(key, '', {
+    expires: new Date(0),
+    path: '/',
+    ...options,
+  })
+
+/**
+ * This is an implementation of {@link Cookies} that mutates the headers of
+ * the given request or response with `Set-Cookie` headers.
+ */
+export class HeaderMutatingCookies extends Cookies {
+  response: HasHeaders
```

Comment:
I renamed this from NextCookies to `HeaderMutatingCookies`. A bit lengthy but I think it makes better sense on why it does that?

I also change the type of `response` from `Response | Request` to `HasHeaders`. This is because I believe we shouldn't pass the entire Response object on production code, as I believe it can cause issues for garbage collection:

```ts
class CustomResponse extends Response {
  constructor(...args) {
    super(...args);
    this.cookies = new HeaderMutatingCookies(this);
    //   ~~~~~~~                             ~~~~
  }
}
```

instead, the better option would be to only pass `headers`, then `HeaderMutatingCookies` and `Response` both target to an object, instead of having circular references.

But I'm not sure about it :neckbeard:

Reply:
Do we want to expose `Cookies` at all? I mean, it kinda makes sense, but maybe we should call this class `Cookies` and the other one `CookieStore` or `CookieMap`?

Reply:
It would make sense to expose a single `Cookie` class (that is, your `HeaderMutatingCookies`).
But then why did we had 2 distinct classes? I could imagine we applied the Single Responsibility Principle, but I failed to see what the actual `Cookie` class is responsible for (in other words, the value it adds, on its own).

If we want to keep them split, I'm up for the `CookieMap` name.

Reply:
yeah the Cookies class make no sense as the subclass does not use the superclass in any way 🤔